src/Sql/Select.php

```diff
@@ -651,7 +651,7 @@ protected function processOrder(PlatformInterface $platform, DriverInterface $dr
                     $v = self::ORDER_ASCENDING;
                 }
             }
-            if (strtoupper($v) == self::ORDER_DESCENDING) {
+            if (strtoupper(trim($v)) == self::ORDER_DESCENDING) {
```

Comment:
No, this is directly from input $select->order('col_name desc') or 'col_name DESC'. That said, case should not matter here just to find if defaults should be applied, so I will switch to case insensitive comparison.